partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/CastTable.kt

```diff
@@ -0,0 +1,701 @@
+package org.partiql.eval.internal.operator.rex
+
+import com.amazon.ionelement.api.ElementType
+import com.amazon.ionelement.api.IonElementException
+import com.amazon.ionelement.api.createIonElementLoader
+import org.partiql.errors.DataException
+import org.partiql.errors.TypeCheckException
+import org.partiql.eval.value.Datum
+import org.partiql.types.PType
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.math.RoundingMode
+
+@Suppress("DEPRECATION")
+internal object CastTable {
```

Comment:
**Could you add a toString() for generating documentation?**

This is great. Nice consolidation of the logic and putting it in a fast data structure.

_Outside the scope..._

I wonder if there are an opportunities to make this part of the datum package in some capacity (or in SPI).

What I'm getting at is for the planner and evaluator to use the same source of truth for casts, and seeing what abstraction we would need.